api/src/test/java/marquez/service/OpenLineageServiceIntegrationTest.java

```diff
@@ -548,6 +554,172 @@ void testJobEvent() throws ExecutionException, InterruptedException {
         .isEqualTo("theDatasource");
   }
 
+  @Test
+  void testStreamingJob() throws ExecutionException, InterruptedException {
+    // (1) Create output
+    LineageEvent.Dataset input =
+        LineageEvent.Dataset.builder().name(DATASET_NAME).namespace(NAMESPACE).build();
+
+    LineageEvent.Dataset output =
+        LineageEvent.Dataset.builder()
+            .name(DATASET_NAME)
+            .namespace(NAMESPACE)
+            .facets(
+                DatasetFacets.builder()
+                    .schema(
+                        new SchemaDatasetFacet(
+                            PRODUCER_URL,
+                            SCHEMA_URL,
+                            Arrays.asList(new SchemaField("col", "STRING", "my name"))))
+                    .build())
+            .build();
+
+    // (2) Streaming job not followed by a COMPLETE event writing to a output
+    UUID firstRunId = UUID.randomUUID();
+    lineageService
+        .createAsync(
+            LineageEvent.builder()
+                .eventType("RUNNING")
+                .run(new LineageEvent.Run(firstRunId.toString(), RunFacet.builder().build()))
+                .job(
+                    LineageEvent.Job.builder()
+                        .name("streaming_job_name")
+                        .namespace(NAMESPACE)
+                        .facets(
+                            JobFacet.builder()
+                                .jobType(
+                                    JobTypeJobFacet.builder()
+                                        .processingType("STREAMING")
+                                        .integration("FLINK")
+                                        .jobType("JOB")
+                                        .build())
+                                .build())
+                        .build())
+                .eventTime(Instant.now().atZone(TIMEZONE))
+                .inputs(Collections.singletonList(input))
+                .outputs(Collections.singletonList(output))
+                .build())
+        .get();
+    Optional<Dataset> datasetRow = datasetDao.findDatasetByName(NAMESPACE, DATASET_NAME);
+
+    // (3) Assert that output is present and has dataset_version written
+    assertThat(datasetRow).isPresent().flatMap(Dataset::getCurrentVersion).isPresent();
+
+    // (4) Assert that job is present although job_version entry is not
```

Comment:
I _think_ a job version for the streaming job should exist based on what's described in the PR?

> It makes sense in this case, to create job version and identify input/output datasets at the beginning of the job and modify it on the fly if a job changes.